README.md

```diff
@@ -1,61 +1,138 @@
+So you built the container successfully!  Congrats.  How can you test it?
 
+You can use the common Docker techniques for launching a container and connecting to it.
 
-## General Repo TODOs:
+The workflow, I adopt is the following:
+
+```bash
+host $ # build the image and name it `example-image`
+host $ docker build -t example-image .
+[...]
+host $ # Start it - but enable ssh and mount a sample test project
+host $ docker run -d \
+      -p 22 -v ~/.ssh/id_rsa.pub:/home/ubuntu/.ssh/authorized_keys \ # To allow for ssh
+      -v hello-world-project:/home/ubuntu/hello-world-project \ # mount a sample project
+      --name example-image-tester \
+      example-image
+5b43b9fd24dd046a389dc2bbc2e84925c011910cfdd2de6cf638dd245f074831
+host $ # Now ssh into the container.  This depends whether you are using Docker natively or through docker-machine
+host $ # With native docker
+host $ CONTAINER_SSH_PORT=$(docker inspect --format='{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}')
+host $ # If using docker-machine, need to connect to the docker-machine ip address
+host $ CONTAINER_SSH_HOST=$(docker-machine ip default) # or set to 127.0.0.1 if using Docker natively on a Linux box
+host $
+host $ ssh -o StrictHostKeyChecking=no -p $CONTAINER_SSH_PORT ubuntu@${CONTAINER_SSH_HOST}
+docker $
+docker $ # now you are in Docker and the build - run any tests you need
+docker $ cd example-image-test
+docker $ # Run any custom test steps you need, e.g. for a node project it will be
+docker $ npm install
+docker $ npm test
+docker $
+docker $ $ also can run any sanity checks, e.g. ensure postgres is running
+ubuntu@22b220709b03:~$ psql
+psql (9.4.5)
+Type "help" for help.
+
+ubuntu=# ;; psql is running
+ubuntu=# ;; psql is running
+ubuntu-# \quit
+```
+
+**TODO**: Offer automated testing, in a similar style to [Docker official image testing](https://github.com/docker-library/official-images/tree/master/test).
+
+# 3. Hooking up CircleCI to use the container
+
+Once you gain confidence in the image you just created, you can start pushing it to new builds.
+You must configure CircleCI Enterprise builders to point to the container
+image.  You can do so, by passing additional environment variable in your
+launch configuration: `CIRCLE_CONTAINER_IMAGE_URI`.
+
+Currently, builders can only support public http or S3 uris (e.g.
+`https://example.com/container_0.0.1.tar.gz` or
+`s3://example/container_0.0.1.tar.gz`.  Future releases will support docker
+URIs (e.g. `docker://acme-org/circleci-test-image:0.0.1`).  For the timebeing,
+we suggest exporting the container and uploading it to S3.
+
+Assuming you have the official aws-cli client, you can export the container and
+upload it to a bucket of your choice.  Ideally, it's located in the same region
+as the builders, and you can reuse the bucket that got created for the CCIE installation:
+
+```bash
+$ ./docker-export example-image > example-image_0.0.1.tar.gz
```

Comment:
Oops - missed pushing it -- it's introduced in https://github.com/circleci/image-builder-prototype/blob/docker-based/docker-export